src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -0,0 +1,416 @@
+package com.phonebookservice.util;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+public class MyArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] data;
+    private int index = 0;
```

Comment:
I see different method has prameter with name "index"

I recommend make it name "currentIndex"